crates/rolldown_common/src/types/stmt_info.rs

```diff
@@ -77,19 +86,48 @@ pub struct StmtInfo {
   /// Top level symbols referenced by this statement.
   pub referenced_symbols: Vec<SymbolRef>,
   pub side_effect: bool,
-  pub is_included: bool,
+  pub included_info: IncludedInfo,
   pub import_records: Vec<ImportRecordId>,
   pub debug_label: Option<String>,
 }
 
 impl StmtInfo {
   pub fn to_debug_stmt_info_for_tree_shaking(&self) -> DebugStmtInfoForTreeShaking {
     DebugStmtInfoForTreeShaking {
-      is_included: self.is_included,
+      is_included: self.partial_included(),
       side_effect: self.side_effect,
       source: self.debug_label.clone().unwrap_or_else(|| "<Noop>".into()),
     }
   }
+
+  pub fn partial_included(&self) -> bool {
```

Comment:
Could we convert
```js
var a, x;
```

to

```
var a;
var x;
```

before scanning since we already need to mutate ast to hoist import statements.

Reply:
I would consider `pre-transpile` is a more genernal and universal way to solve many tree-shaking edge cases.

For example

we could deal with the following case by transpile
```js
{
  var a;
  var b;
}

export { a, b }
```

to

```
{
var a;
var b;

export {a, b }
}
```

without changing how tree_shaking works currently.

Reply:
The whole problem is that the current esbuild's treeshaking implementation works on statements level.

I understand that introducing `partial_included` and `fully_included` could solve the multiple declaration case, but it requires changes on many places, linking, code splitting, deconflicting and finalizing. It's kind of unworthy.

The `pre-transpile` way solve the problem by only need to add extra code in transpile step without touching any other places.

Reply:
However, I would consider `pre-transpile` way is kind of workaround. If we could design a treeshaking implementation that's works on the level that smaller than statements, I would not suggest using `pre-transpile`.

In short, I would suggest using `pre-transpile` way on this case for patching esbuid's treeshaking implementation.

If we gonna design a new treeshaking implementation, we could introducing new concepts to solve the problem.

Reply:
I would consider the `pre-transpile` is a another representation of `AST -> HIR`, while we are `AST -> (Tree shaking friendly's) AST`.

Reply:
Yeah, but transpile would introduce another source map remapping, which should be another overhead, I will try to find another way to finish this with minimal modification.